### DISCUSSION TITLE
pages apart from index.html dont connect to socket .io

### DISCUSSION THREAD

**Discussion by flippedoffbit** [socketio/socket.io#4544] [Q&A] [answered]
i was mixing and matching few example in docs
when i tried accessing socket.io through pages other then index.html 
it just logs errors in console 

WebSocket connection to 'ws://localhost:4000/socket.io/?EIO=4&transport=websocket&sid=Xr6J7yHGyTYv75bzAAA6' failed: WebSocket is closed before the connection is established

sources that i am working with are at 
https://github.com/acnologia000/sockchat

**Comment by darrachequesne** [comment, marked as answer]
It seems there is a typo in the [agent.html](https://github.com/acnologia000/sockchat/blob/35f83ebf27ad3d95fec6c659ca9e91e7979bca61/static/agent.html#L79) file:

```js
socket.auth = { username }
```



Instead of:

```js
socket.auth = { user_name }
```

in the [index.html](https://github.com/acnologia000/sockchat/blob/35f83ebf27ad3d95fec6c659ca9e91e7979bca61/static/index.html#L80)

So the middleware [here](https://github.com/acnologia000/sockchat/blob/35f83ebf27ad3d95fec6c659ca9e91e7979bca61/index.js#L15-L18) prevents the connection and closes the WebSocket connection:

```js
const username = socket.handshake.auth.user_name;
if (!username) {
  return next(new Error("invalid username"));
}
```

Could you please check?

**Reply by flippedoffbit** [reply]
thanks for the response, i was making such a silly mistake
thanks for your time again
